sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -75,6 +75,18 @@ private[sql] object JDBCRelation {
     }
     ans.toArray
   }
+
+  def getEffectiveProperties(
+      connectionProperties: Properties,
+      extraOptions: scala.collection.Map[String, String] = Map()): Properties = {
+    val props = new Properties()
+    extraOptions.foreach { case (key, value) =>
```

Comment:
Sorry, only now really thinking through this -- if you import `scala.collection.JavaConverters._` then this clause can just be `props.putAll(extraOptions.asJava)`